astroid/nodes/node_ng.py

```diff
@@ -323,7 +322,19 @@ def frame(
 
         :returns: The first parent frame node.
         """
-        return self.parent.frame()
+        if self.parent is None:
+            if future:
+                raise ParentMissingError(target=self)
+            warnings.warn(
+                "In astroid 3.0.0 NodeNG.frame() will return either a Frame, "
```

Comment:
```suggestion
                "In astroid 3.0.0 NodeNG.frame() will return either a Frame node, "
```
Technically those could be `FunctionDef`, `Module`, `ClassDef`, and `Lambda`. `Frame node` is probably enough though.